cdap-common/src/main/java/co/cask/cdap/common/conf/CConfigurationUtil.java

```diff
@@ -33,4 +35,18 @@ public static void copyTxProperties(CConfiguration cConf, org.apache.hadoop.conf
       }
     }
   }
+
+  public static void checkCConfValidity(CConfiguration cConf) {
+    // Checks to ensure that certain keys (e.g. "root.prefix") are valid as expected by CDAP.
+    checkAlphaNumberic(cConf, Constants.CFG_ROOT_NAMESPACE);
+    checkAlphaNumberic(cConf, Constants.Dataset.TABLE_PREFIX);
```

Comment:
No where in the code had direct access to the root namespace before (only through things that inherited it, such as Dataset Table Prefix). I didn't find validations on table prefix.
So I think this validation on the two properties are not duplicating logic.